lib/weatherApp.js

```diff
@@ -0,0 +1,46 @@
+(function (factory) {
+    if (typeof define === 'function' && define['amd']) {
+        define(['knockout'], factory);
+    } else if (typeof require === 'function' && typeof exports === 'object' && typeof module === 'object') {
+        module.exports = factory(require('knockout'));
+    } else {
+        window.WeatherApp = factory(ko);
+    }
+} (function (ko) {
+    return function WeatherApp() {
+        'use strict';
+        let city = ko.observable("");
+        let api = "http://api.openweathermap.org/data/2.5/weather?q=";
+        let unit = "&units=metric&";
+        let apiId = "appid=bd959553bce6759da749a8bcad48f038";
+        let maxTemp = ko.observable("");
+        let minTemp = ko.observable("");
+        let isTempVisible = ko.observable(false);
+        let josnCityName = ko.observable("");
+
+        function showWeather() {
+            let url = api + city() + unit + apiId;
```

Comment:
Currently, users can click the Submit button several times in a row before the information has actually come back. It would be good to disable the submit button as soon as the user clicks it, and then re-enable it when the data comes back (or if the request failed).